### PR TITLE
Upgrade algod and indexer images to Go 1.17

### PR DESCRIPTION
**Pull request by michaeldiamant**
Updates algod and indexer images to support Go v1.17.  

Relates to these PRs:
* https://github.com/algorand/go-algorand/pull/3919
* https://github.com/algorand/indexer/pull/980

### CODE REVIEW

**Comment by michaeldiamant** [comment]
Testing via https://github.com/algorand/java-algorand-sdk/pull/319#issuecomment-1125210395 worked as expected.  Marking PR ready for review.

File: docker/algod/source/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.16.11
+FROM golang:1.7
```
Comment (michaeldiamant) [inline docker/algod/source/Dockerfile:1]:
```suggestion
FROM golang:1.17
```

I made a typo here.

**Review by algoidurovic — APPROVED** [review]
LGTM. Maybe it would be worthwhile to look into why the golang version typo wasn't causing a test failure.

**Comment by michaeldiamant** [comment]
> why the golang version typo wasn't causing a test failure.

* `docker/algod/source/Dockerfile` isn't referenced by `docker-compose.yml`, which explains the absence of downstream build failure.
* Locally, I confirmed the PR builds successfully via `docker build -f docker/algod/source/Dockerfile .`.

**Review by winder — APPROVED** [review]
LGTM

**Pull request merged**